python/lsst/daf/butler/registry/tests/_registry.py

```diff
@@ -2236,3 +2234,75 @@ def testBindInQueryDatasets(self):
             set(registry.queryDatasets("flat", band="r", collections=...)),
             set(registry.queryDatasets("flat", where="band=my_band", bind={"my_band": "r"}, collections=...)),
         )
+
+    def testQueryResultSummaries(self):
+        """Test summary methods like `count`, `any`, and `explain_no_results`
+        on `DataCoordinateQueryResults` and `DatasetQueryResults`
+        """
+        registry = self.makeRegistry()
+        self.loadData(registry, "base.yaml")
+        self.loadData(registry, "datasets.yaml")
+        # Default test dataset has two collections, each with both flats and
+        # biases.  Add a new collection with only biases.
+        registry.registerCollection("biases", CollectionType.TAGGED)
+        registry.associate("biases", registry.queryDatasets("bias", collections=["imported_g"]))
+        # First query yields two results, and involves no postprocessing.
+        query1 = registry.queryDataIds(["physical_filter"], band="r")
+        self.assertTrue(query1.any(execute=False, exact=False))
+        self.assertTrue(query1.any(execute=True, exact=False))
+        self.assertTrue(query1.any(execute=True, exact=True))
+        self.assertEqual(query1.count(exact=False), 2)
```

Comment:
Should it be `assertGreaterEqual`?

Reply:
This test is constructed such that equality is safe.  That makes it a bit more dependent on the implementation than I'd normally like in a test, but I think it's appropriate in this case to make sure I'm testing all the code paths (in other words, if the implementation changed enough to break this, the tests really should be modified).